pkg/pkgmgmt/search.go

```diff
@@ -0,0 +1,56 @@
+package pkgmgmt
+
+import (
+	"encoding/json"
+	"sort"
+	"strings"
+
+	"github.com/gobuffalo/packr/v2"
+	"github.com/pkg/errors"
+)
+
+// Searcher contains a packr.Box containing a searchable list of packages
+type Searcher struct {
+	Box *packr.Box
+}
+
+// NewSearcher returns a Searcher with the provided packr.Box
+func NewSearcher(box *packr.Box) Searcher {
+	return Searcher{
+		Box: box,
+	}
+}
+
+// Search searches for packages matching the optional provided name,
+// returning the full list if none is provided
+func (s *Searcher) Search(name, pkgType string) (PackageList, error) {
+	data, err := s.Box.Find("index.json")
+	if err != nil {
+		return PackageList{}, errors.Wrapf(err, "error loading %s list\n", pkgType)
+	}
+
+	var pl PackageList
+	err = json.Unmarshal(data, &pl)
+	if err != nil {
+		return PackageList{}, errors.Wrapf(err, "could not parse %s list\n", pkgType)
+	}
+
+	if name == "" {
+		sort.Sort(pl)
+		return pl, nil
+	}
+
+	results := PackageList{}
+	for _, p := range pl {
+		if strings.HasPrefix(p.Name, name) {
```

Comment:
Sorry for not asking the first time around. Neither of these are requests, just thinking out loud about how the search command should behave and am looking for your opinion.

1. Are we sure we want to do a case-sensitive comparison? It may be helpful to use ToLower first and compare when doing a search.

2. Would `string.Contains` be more intuitive for a search than prefix?

Reply:
1. Had not thought of this.  I like your idea on sanitizing the name via toLower before search. 👍 
2. Good detail to call out. I had waffled on this; first Contains then HasPrefix.  Thinking on it more, indeed, we wouldn't want to miss, say, a `hashicorp-vagrant` mixin if searching `vagrant`.  I'll switch back to `Contains`.